var/spack/repos/builtin/packages/py-pyside/package.py

```diff
@@ -33,17 +33,23 @@ class PyPyside(PythonPackage):
         preferred=True,
     )
 
-    depends_on("cmake", type="build")
-
+    # to prevent error: 'PyTypeObject' {aka 'struct _typeobject'} has no member
+    # named 'tp_print'
+    depends_on("python@:3.8", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     # in newer pip versions --install-option does not exist
     depends_on("py-pip@:23.0", type="build")
+    depends_on("cmake", type="build")
+
     depends_on("py-sphinx", type=("build", "run"))
     depends_on("py-sphinx@:3.5.0", type=("build", "run"), when="@:1.2.2")
     depends_on("qt@4.5:4.9")
     depends_on("libxml2@2.6.32:")
     depends_on("libxslt@1.1.19:")
 
+    # to prevent ImportError: cannot import name 'soft_unicode' from 'markupsafe'
+    conflicts("^py-markupsafe@2.0.2:")
```

Comment:
This package doesn't import markupsafe. Does this belong in some other package?

Reply:
Same as for `py-shiboken`, `py-markupsafe` comes from `py-jinja` which comes from `py-sphinx`

without `py-markupsafe` conflict
```
[+]      ^py-sphinx@7.0.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
[+]          ^py-jinja2@3.1.2%gcc@12.3.1~i18n build_system=python_pip arch=linux-fedora37-x86_64
[+]              ^py-markupsafe@2.1.3%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
```

with `py-markupsafe` conflict
```
[+]      ^py-sphinx@7.0.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
[+]          ^py-jinja2@3.1.2%gcc@12.3.1~i18n build_system=python_pip arch=linux-fedora37-x86_64
[+]              ^py-markupsafe@2.0.1%gcc@12.3.1 build_system=python_pip arch=linux-fedora37-x86_64
...
```

Reply:
Ok, I found [this issue for jinja](https://github.com/pallets/jinja/issues/1585), so I think we can add 
```
conflicts("^py-markupsafe@2.0.2:", when="@:2.11.3")
```
to `py-jinja`.
The problem is, that it's hard to test. `py-jinja@2.11.3` installs fine without the conflict even with `--test=root` for me.
I will open a PR for it.

Reply:
Done in #38913 and I have also removed the conflict here.